extensions/amp-analytics/0.1/test/test-amp-analytics.js

```diff
@@ -448,6 +451,18 @@ describes.realWin('amp-analytics', {
     });
   });
 
+  it('does send a hit when parentPostMessage is provided inabox', function() {
+    env.win.AMP_MODE.runtime = 'inabox';
+    const analytics = getAnalyticsTag({
+      'requests': {'foo': 'https://example.com/bar'},
+      'triggers': [{'on': 'visible', 'parentPostMessage': 'foo'}],
+    });
+    return waitForNoSendRequest(analytics).then(() => {
+      expect(sendRequestSpy).to.have.not.been.called;
+      expect(postMessageSpy).to.have.been.called;
+    });
+  });
+
```

Comment:
Could we have a test for when there's both a post message and a request?

Reply:
added test
inabox case: w/ request and postMessage, w/o request postMessage, only postMessage
non inabox case: with postMessage and request